File: content/snippets/python/interview_lcci_01_09_string_rotation.py
def is_string_rotation(s1: str, s2: str) -> bool:
    return len(s1) == len(s2) and s2 in (s1 + s1)


def main() -> None:
    print(f"waterbottle vs erbottlewat -> {is_string_rotation('waterbottle', 'erbottlewat')}")
    print(f"aa vs aba -> {is_string_rotation('aa', 'aba')}")


if __name__ == "__main__":
    main()

File: content/snippets/typescript/interview_lcci_01_09_string_rotation.ts
function isStringRotation(s1: string, s2: string): boolean {
  return s1.length === s2.length && (s1 + s1).includes(s2);
}

function main(): void {
  console.log(`waterbottle vs erbottlewat -> ${isStringRotation("waterbottle", "erbottlewat")}`);
  console.log(`aa vs aba -> ${isStringRotation("aa", "aba")}`);
}

main();
